qt/python/mantidqt/mantidqt/widgets/test/test_fitpropertybrowser.py

```diff
@@ -132,6 +132,40 @@ def test_fit_result_workspaces_are_added_to_browser_when_fitting_done(self):
         self.assertEqual(name + "_Parameters", workspaceList.item(1).text())
         self.assertEqual(name + "_Workspace", workspaceList.item(2).text())
 
+    def test_fitting_done_follows_ads_clear_fitting_done_another(self):
+        name_1 = "ws_1"
+        fig, canvas, _ = self._create_and_plot_matrix_workspace(name_1)
+        property_browser = self._create_widget(canvas=canvas)
+        property_browser.setOutputName(name_1)
+
+        # create fake fit output results
+        matrixWorkspace = WorkspaceFactory.Instance().create("Workspace2D", NVectors=3, YLength=5, XLength=5)
+        tableWorkspace = WorkspaceFactory.createTable()
+        AnalysisDataService.Instance().addOrReplace(name_1 + "_Workspace", matrixWorkspace)
+        AnalysisDataService.Instance().addOrReplace(name_1 + "_Parameters", tableWorkspace)
+        AnalysisDataService.Instance().addOrReplace(name_1 + "_NormalisedCovarianceMatrix", tableWorkspace)
+
+        property_browser.fitting_done_slot(name_1 + "_Workspace")
+        workspaceList = property_browser.getWorkspaceList()
+        self.assertEqual(3, workspaceList.count())
+
+        AnalysisDataService.clear()
```

Comment:
Not sure about running `ADS.clear` in the middle of a test, I know that there can be issues with workspaces of the same name shared between tests when they are run concurrently - is it possible to delete the specific workspaces rather than clear here?

Reply:
Thanks for noting that! I have just updated the unit test with that.